internal/generator/ecdsa/template/ecdsa.go.tmpl

```diff
@@ -211,16 +211,21 @@ func (privKey *PrivateKey) Sign(message []byte, hFunc hash.Hash) ([]byte, error)
 		}
 		s.Mul(r, scalar)
 
-		// compute the hash of the message as an integer
-		dataToHash := make([]byte, len(message))
-		copy(dataToHash[:], message[:])
-		hFunc.Reset()
-		_, err := hFunc.Write(dataToHash[:])
-		if err != nil {
-			return nil, err
-		}
-		hramBin := hFunc.Sum(nil)
-		m := HashToInt(hramBin)
+		m := new(big.Int)
```

Comment:
It is sufficient here to do `var m *big.Int` and not allocate. You assign value to m later. This allocation will be wasted.